src/audio/google/google_ctc_audio_processing_mock.c

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2024 Google LLC.
+//
+// Author: Eddy Hsu <eddyhsu@google.com>
+#include "google_ctc_audio_processing.h"
+#include "google_ctc_audio_processing_sof_message_reader.h"
+
+#include <rtos/alloc.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "ipc/topology.h"
+
+struct GoogleCtcAudioProcessingState {
+};
+
+GoogleCtcAudioProcessingState *GoogleCtcAudioProcessingCreate(void)
+{
+	struct GoogleCtcAudioProcessingState *s =
+		rballoc(0, SOF_MEM_CAPS_RAM, sizeof(GoogleCtcAudioProcessingState));
+	if (!s)
+		return NULL;
+
+	return s;
+}
+
+GoogleCtcAudioProcessingState *GoogleCtcAudioProcessingCreateWithConfig(int num_frames,
+									int partition_size,
+									int impulse_size,
+									int chunk_size,
+									int sample_rate,
+									int is_symmetric,
+									const uint8_t *config,
+									int config_size)
+{
+	struct GoogleCtcAudioProcessingState *s =
+		rballoc(0, SOF_MEM_CAPS_RAM, sizeof(GoogleCtcAudioProcessingState));
+	if (!s)
+		return NULL;
+
+	return s;
+}
+
+void GoogleCtcAudioProcessingFree(GoogleCtcAudioProcessingState *state)
+{
+	rfree(state);
+}
+
+void GoogleCtcAudioProcessingProcess(GoogleCtcAudioProcessingState *state,
+				     const float *src, float *dest,
+				     int num_frames, int num_channels)
+{
+	memcpy_s(dest, sizeof(float) * num_frames * num_channels,
+		 src, sizeof(float) * num_frames * num_channels);
+}
+
+void GoogleCtcAudioProcessingParseSofConfigMessage(uint8_t *message,
+						   size_t message_size,
+						   uint8_t **config,
+						   size_t *config_size,
+						   bool *config_present)
+{
+	*config = NULL;
```

Comment:
leak?

Reply:
I think it's irrelevant in a mock function.